### PR TITLE
refactor(ci/github): run weaver tests on schedule, re-use workflows

### PR DESCRIPTION
**Pull request by jenniferlianne**
**Primary change:**

Creates a central workflow to run all 13 weaver test workflows from either a
push, a pull request, a github command, or a schedule.

Passes a 'run_all' boolean to sub-workflows to specify if all tests should be run, not
just those with changes.

**Secondary changes:**
- update concurrency groups to allow triggering from central workflow
- remove ability to run the 13 workflows individually to declutter ci menu

**Notes:**
* For example workflow output see here: https://github.com/jenniferlianne/cacti/actions/runs/10062422329  (jobs are displayed as a tree on the left).  
* A number of weaver tests are commented out with an 'if false' these were left unchanged.

Depends on #3446

**Pull Request Requirements**
- [x] Rebased onto `upstream/main` branch and squashed into single commit to help maintainers review it more efficient and to avoid spaghetti git commit graphs that obfuscate which commit did exactly what change, when and, why.
- [x] Have git sign off at the end of commit message to avoid being marked red. You can add `-s` flag when using `git commit` command. You may refer to this [link](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more information.
- [x] Follow the Commit Linting specification. You may refer to this [link](https://www.conventionalcommits.org/en/v1.0.0-beta.4/#specification) for more information. 

**Character Limit**
- [x ] Pull Request Title and Commit Subject must not exceed 72 characters (including spaces and special characters).
- [ x] Commit Message per line must not exceed 80 characters (including spaces and special characters).

**A Must Read for Beginners**
For rebasing and squashing, here's a [must read guide](https://github.com/servo/servo/wiki/Beginner's-guide-to-rebasing-and-squashing) for beginners.

### CODE REVIEW

**Review by petermetz — APPROVED** [review]

File: .github/workflows/test_weaver.yaml
Comment (sandeepnRES) [inline .github/workflows/test_weaver.yaml:11]: Hi @jenniferlianne Thanks for the PR it looks really organized now, but one doubt, what does this `allow_running_weaver_tests_if_code_unchanged` in `push: branches` do?

**Review by sandeepnRES — APPROVED** [review]
LGTM, just one optional change if you want to, i.e. rename `Test  Weaver` to `Weaver_CI` , just like `Cactus_CI` just to be consistent in naming.
But this is optional just aesthetic, I don't have much issue with current one also.

File: .github/workflows/test_weaver.yaml
Comment (jenniferlianne) [reply .github/workflows/test_weaver.yaml]: This was the branch name in my repo.  Have removed this reference and updated the name of the combined test file.  

I notice that the PR ActionLint fails to say that the files weaver workflow files are not found, however the weaver tests are running fine so I think this is a false positive.

**Review by petermetz — COMMENTED** [review]
@jenniferlianne Agreed, the actionlint looks bugged out. Could you please try to upgrade the actionlint version to the current latest instead of whatever we are using at the moment? I'm hoping that this bug was discovered and fixed a while back!

Also plus one for the rename that @sandeepnRES mentioned above.

**Review by outSH — APPROVED** [review]

**Comment by sandeepnRES** [comment]
I thin its because the required CI checks have been moved under `Weaver_CI` that's why github is unable to find those and blocking the PR, what I'll do is remove those required checks for now, and after merging the PR add the new ones.

**Comment by sandeepnRES** [comment]
I've removed all weaver checks from required ones from branch protection rules (an exception for this PR as it is refactoring the CI itself).
@jenniferlianne ActionLint is failing, please investigate, once the PR is merged please ping me I'll add new CI checks as required in branch protection rules.

File: .github/workflows/test_weaver.yaml
Comment (sandeepnRES) [reply .github/workflows/test_weaver.yaml]: I think its because the required CI checks have been moved under Weaver_CI that's why github is unable to find those and blocking the PR, what I'll do is remove those required checks for now, and after merging the PR add the new ones.

**Comment by jenniferlianne** [comment]
@petermetz I have updated the ActionLint version to the latest, but it still fails.  The linter might be a getting stuck on a permissions issue between repos.  (Settings->General->Actions Permissions).  Mine is set to run all.

**Comment by petermetz** [comment]
@jenniferlianne Now I remember! We didn't have time to fix all the linter errors in certain yaml files at the introduction of the ActionLint job and there was no documented way to ignore them so the ActionLint job deletes a list of the yaml files that we wanted to ignore as a workaround. I've just pushed a potential fix, but let's see if it actually works.

**Comment by petermetz** [comment]
@jenniferlianne I've put the fix here: #3446 once that gets merged to main and this one rebased onto main then the ActionLint job will start passing on this PR as well.

**Pull request merged**
